scripts/un/sdg/process.py

```diff
@@ -27,54 +27,51 @@
 Usage: python3 process.py
 '''
 import collections
+import csv
+import math
 import os
 import pandas as pd
 import shutil
 import sys
 
+from string import punctuation
+
 sys.path.append(
     os.path.dirname(os.path.dirname(os.path.dirname(
         os.path.abspath(__file__)))))
 from un.sdg import util
 
 
-def get_geography(code, type):
-    '''Returns dcid of geography.
+def get_place_mappings(file):
+    '''Produces map of SDG code -> dcid:
 
     Args:
-        code: Geography code.
-        type: Geography type.
+      file: Input file path.
 
     Returns:
-        Geography dcid.
+      Map of SDG code -> dcid:
     '''
-
-    # Currently only support Country, City, and select Regions .
-    if code in util.REGIONS:
-        return 'dcs:' + util.REGIONS[code]
-    elif type == 'Country' and code in util.PLACES:
-        return 'dcs:country/' + util.PLACES[code]
-    elif type == 'City':
-        # Remove country prefix for now.
-        city = '_'.join(code.split('_')[1:])
-        if city in util.CITIES and util.CITIES[city]:
-            return 'dcs:' + util.CITIES[city]
-    return ''
+    place_mappings = {}
+    with open(file) as f:
+        reader = csv.DictReader(f)
+        for row in reader:
+            place_mappings[str(row['sdg'])] = str(row['dcid'])
+    return place_mappings
 
 
-def get_unit(units, base_period):
-    '''Returns dcid of unit.
+def get_geography(code, place_mappings):
+    '''Returns dcid of geography.
 
     Args:
-        unit: Unit.
-        base_period: Base period of unit.
+        code: Geography code.
+        place_mappings: Map of SDG code -> dcid.
 
     Returns:
-        Unit dcid.
+        Geography dcid.
     '''
-    if util.is_valid(base_period):
-        return f'[{units} {base_period}]'
-    return 'dcs:SDG_' + units
+    if str(code) in place_mappings:
+        return 'dcs:' + place_mappings[str(code)]
```

Comment:
nit: place references should be `dcid:`?

Reply:
done